x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/InferenceToXContentCompressor.java

```diff
@@ -45,33 +49,33 @@ private InferenceToXContentCompressor() {}
     static <T> T inflate(String compressedString,
                          CheckedFunction<XContentParser, T, IOException> parserFunction,
                          NamedXContentRegistry xContentRegistry) throws IOException {
-        try(XContentParser parser = XContentHelper.createParser(xContentRegistry,
+        try(XContentParser parser = JsonXContent.jsonXContent.createParser(xContentRegistry,
             LoggingDeprecationHandler.INSTANCE,
-            inflate(compressedString, MAX_INFLATED_BYTES),
-            XContentType.JSON)) {
+            inflate(compressedString, MAX_INFLATED_BYTES))) {
             return parserFunction.apply(parser);
         }
     }
 
     static Map<String, Object> inflateToMap(String compressedString) throws IOException {
         // Don't need the xcontent registry as we are not deflating named objects.
-        try(XContentParser parser = XContentHelper.createParser(NamedXContentRegistry.EMPTY,
+        try(XContentParser parser = JsonXContent.jsonXContent.createParser(NamedXContentRegistry.EMPTY,
             LoggingDeprecationHandler.INSTANCE,
-            inflate(compressedString, MAX_INFLATED_BYTES),
-            XContentType.JSON)) {
+            inflate(compressedString, MAX_INFLATED_BYTES))) {
             return parser.mapOrdered();
         }
     }
 
-    static BytesReference inflate(String compressedString, long streamSize) throws IOException {
+    static InputStream inflate(String compressedString, long streamSize) throws IOException {
         byte[] compressedBytes = Base64.getDecoder().decode(compressedString.getBytes(StandardCharsets.UTF_8));
+        // If the compressed length is already too large, it make sense that the inflated length would be as well
```

Comment:
This comment isn't true in general.  If you compress a very small string then the compressed size is bigger than the original.  For example `echo a | gzip -9 | wc -c` returns 22.

The assumption is OK with the sort of `streamSize` values this method is going to be called with given the current code, so it's not essential to change now, but you could make it something like `if (compressedBytes.length > Math.max(100L, streamSize))` in case it ever needs to cope with an extreme edge case in the future.

Also, it would be good to adjust the comment to acknowledge the edge case.